File: frontend/src/microservices/content/objects/UploadReviewImagePlannerRequest.ts
// UploadReviewImagePlannerRequest：content 域评论图片上传请求对象。

export type UploadReviewImagePlannerRequest = {
  userId: string
  originalFileName: string
  contentType: string
  base64Content: string
}

export const uploadReviewImagePlannerRequestFromJson = (json: string): UploadReviewImagePlannerRequest =>
  JSON.parse(json) as UploadReviewImagePlannerRequest

export const uploadReviewImagePlannerRequestToJson = (value: UploadReviewImagePlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/content/objects/UploadReviewImagePlannerRequest.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UploadReviewImagePlannerRequest(
    userId: String,
    originalFileName: String,
    contentType: String,
    base64Content: String
)
object UploadReviewImagePlannerRequest:
  given sourceEncoder: Encoder[UploadReviewImagePlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[UploadReviewImagePlannerRequest] = deriveDecoder
